wire-compiler/src/main/java/com/squareup/wire/WireCompiler.java

```diff
@@ -228,9 +248,23 @@ void compile() throws IOException {
       }
     }
 
+    LinkedHashMap<ProtoType, TypeName> protoTypeToJavaType = new LinkedHashMap<>();
+    LinkedHashMap<ProtoType, TypeName> protoTypeToAdapterClass = new LinkedHashMap<>();
```

Comment:
Can you change the left hand side to be a `Map` instead of a `LinkedHashMap` ? `LinkedHashMap` is an implementation detail!